source/common/http/filter/fault_filter.h

```diff
@@ -82,14 +93,23 @@ class FaultFilter : public StreamDecoderFilter {
   void setDecoderFilterCallbacks(StreamDecoderFilterCallbacks& callbacks) override;
 
 private:
+  void recordAbortsInjectedStats();
+  void recordDelaysInjectedStats();
   void resetTimerState();
   void postDelayInjection();
   void abortWithHTTPStatus();
-  bool matchesTargetCluster();
+  bool matchesTargetUpstreamCluster();
+  bool matchesDownstreamNodes(const HeaderMap& headers);
 
   FaultFilterConfigSharedPtr config_;
   StreamDecoderFilterCallbacks* callbacks_{};
   Event::TimerPtr delay_timer_;
+  std::string downstream_cluster_{};
+
+  std::string delay_percent_key_{"fault.http.delay.fixed_delay_percent"};
```

Comment:
those could not be const in current impl because it depends on whether downstream cluster specific settings are used of not.